Backend/DataAccessLayer/Repositories/OperationRepository.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using DataAccessLayer.Models;
+using System.Data.Entity;
+using DataAccessLayer.Database;
+
+namespace DataAccessLayer.Repositories
+{
+    public class OperationRepository
+    {
+
+        public int createService(Service service, DatabaseContext _db)
+        {
+            try
+            {
+                _db.Services.Add(service);
+                return 1;
+
+            } 
+            catch(Exception)
+            {
+                return 0;
+            }
+        }
+
+        public int createService(string serviceName, DatabaseContext _db)
+        {
+            try
+            {
+                Service service = new Service
+                {
+                    ServiceName = serviceName,
+                    Disabled = false
+                };
+
+                _db.Services.Add(service);
+                return 1;
+
+            }
+            catch (Exception)
+            {
+                return 0;
+            }
+        }
+
+        public int disableService(Guid id, DatabaseContext _db)
+        {
+            try
+            {
+                Service service = _db.Services.Find(id);
+                //service.Disabled = true;
+                //service.UpdatedAt = DateTime.UtcNow;
+                //_db.Entry(service).State = EntityState.Modified;
+                return 1;
+            }
+            catch(Exception)
+            {
+                return 0;
+            }
+        }
+
+        public int enableService(Guid id, DatabaseContext _db)
+        {
+            try
+            {
+                Service service = _db.Services
+                    .Where(s => s.Id == id)
+                    .FirstOrDefault();
+                service.Disabled = false;
+                service.UpdatedAt = DateTime.UtcNow;
+                _db.Entry(service).State = EntityState.Modified;
+                return 1;
+            }
+            catch (Exception)
+            {
+                return 0;
+            }
+        }
+
+        public int deleteService(Guid id, DatabaseContext _db)
+        {
+            try
+            {
+                var service = _db.Services.Find(id);
+                if (service == null)
+                    return 0;
+                _db.Entry(service).State = EntityState.Deleted;
+                return 1;
+            }
+            catch(Exception)
+            {
+                return 0;

```

Comment:
Problem with your design here:

If the service is not found, the method returns 0. If the method encounters an error, it returns 0.

From the standpoint of someone trying to use this method,  this is very confusing.

1. There is no way to differentiate between an error and "not found"
2. There is no way to get an accurate error message for what failed.

Perhaps handling errors within our services may not be the best idea. Alternatively, you could return a negative value if an exception was encountered.

Reply:
Decided to go with the following approach:
1 - Operation completed
0 - Error with database operation
-1 - Incomplete object information (I.E. ServiceName = Null)